### PR TITLE
Add Rust 1.20.0 images

### PR DESCRIPTION
**Pull request by sfackler**

### CODE REVIEW

**Comment by yosifkit** [comment]
<details>
<summary>Diff:</summary>

```diff
diff --git a/_bashbrew-arches b/_bashbrew-arches
index c576c8d..62c61bd 100644
--- a/_bashbrew-arches
+++ b/_bashbrew-arches
@@ -1,3 +1,11 @@
+rust:1.19 @ amd64
+rust:1.19 @ arm32v7
+rust:1.19 @ arm64v8
+rust:1.19 @ i386
+rust:1.19-jessie @ amd64
+rust:1.19-jessie @ arm32v7
+rust:1.19-jessie @ arm64v8
+rust:1.19-jessie @ i386
 rust:jessie @ amd64
 rust:jessie @ arm32v7
 rust:jessie @ arm64v8
diff --git a/_bashbrew-list b/_bashbrew-list
index c33d7fd..9329564 100644
--- a/_bashbrew-list
+++ b/_bashbrew-list
@@ -7,6 +7,12 @@ rust:1.19-stretch
 rust:1.19.0
 rust:1.19.0-jessie
 rust:1.19.0-stretch
+rust:1.20
+rust:1.20-jessie
+rust:1.20-stretch
+rust:1.20.0
+rust:1.20.0-jessie
+rust:1.20.0-stretch
 rust:jessie
 rust:latest
 rust:stretch
diff --git a/rust_jessie/Dockerfile b/rust_1.19-jessie/Dockerfile
similarity index 100%
copy from rust_jessie/Dockerfile
copy to rust_1.19-jessie/Dockerfile
diff --git a/rust_latest/Dockerfile b/rust_1.19/Dockerfile
similarity index 100%
copy from rust_latest/Dockerfile
copy to rust_1.19/Dockerfile
diff --git a/rust_jessie/Dockerfile b/rust_jessie/Dockerfile
index c917344..69e78a8 100644
--- a/rust_jessie/Dockerfile
+++ b/rust_jessie/Dockerfile
@@ -9,18 +9,18 @@ RUN set -eux; \
 # this "case" statement is generated via "update.sh"
     dpkgArch="$(dpkg --print-architecture)"; \
 	case "${dpkgArch##*-}" in \
-		amd64) rustArch='x86_64-unknown-linux-gnu'; rustupSha256='7b5ce33a881992b285e2aa6cbc785da4138c5bab7c8c9b55c06918bfb1ba0efa' ;; \
-		armhf) rustArch='armv7-unknown-linux-gnueabihf'; rustupSha256='a92b003a15b2e4bd240c0f1d46232958c173f5605814e19961fc8a4d99a25b3e' ;; \
-		arm64) rustArch='aarch64-unknown-linux-gnu'; rustupSha256='9029bf864848ab313a2c6806492ae471d1578c8196acdb220e71d6a785241a5a' ;; \
-		i386) rustArch='i686-unknown-linux-gnu'; rustupSha256='4a478c977b7b4900456c2d4dd165019e7c923ebdaba3f47316717d1690387d9a' ;; \
+		amd64) rustArch='x86_64-unknown-linux-gnu'; rustupSha256='f5833a64fd549971be80fa42cffc6c5e7f51c4f443cd46e90e4c17919c24481f' ;; \
+		armhf) rustArch='armv7-unknown-linux-gnueabihf'; rustupSha256='67a98a67f7f7bf19c5cde166499acb8299f2f8fa88c155093df53b66da1f512a' ;; \
+		arm64) rustArch='aarch64-unknown-linux-gnu'; rustupSha256='82fe368c4ebf1683d57e137242793a4417042639aace8bd514601db7d79d3645' ;; \
+		i386) rustArch='i686-unknown-linux-gnu'; rustupSha256='7a1c085591f6c1305877919f8495c04a1c97546d001d1357a7a879cedea5afbb' ;; \
 		*) echo >&2 "unsupported architecture: ${dpkgArch}"; exit 1 ;; \
 	esac; \
     \
-    url="https://static.rust-lang.org/rustup/archive/1.5.0/${rustArch}/rustup-init"; \
+    url="https://static.rust-lang.org/rustup/archive/1.6.0/${rustArch}/rustup-init"; \
     wget "$url"; \
     echo "${rustupSha256} *rustup-init" | sha256sum -c -; \
     chmod +x rustup-init; \
-    ./rustup-init -y --no-modify-path --default-toolchain 1.19.0; \
+    ./rustup-init -y --no-modify-path --default-toolchain 1.20.0; \
     rm rustup-init; \
     chmod -R a+w $RUSTUP_HOME $CARGO_HOME; \
     rustup --version; \
diff --git a/rust_latest/Dockerfile b/rust_latest/Dockerfile
index 99e4633..13a231a 100644
--- a/rust_latest/Dockerfile
+++ b/rust_latest/Dockerfile
@@ -9,18 +9,18 @@ RUN set -eux; \
 # this "case" statement is generated via "update.sh"
     dpkgArch="$(dpkg --print-architecture)"; \
 	case "${dpkgArch##*-}" in \
-		amd64) rustArch='x86_64-unknown-linux-gnu'; rustupSha256='7b5ce33a881992b285e2aa6cbc785da4138c5bab7c8c9b55c06918bfb1ba0efa' ;; \
-		armhf) rustArch='armv7-unknown-linux-gnueabihf'; rustupSha256='a92b003a15b2e4bd240c0f1d46232958c173f5605814e19961fc8a4d99a25b3e' ;; \
-		arm64) rustArch='aarch64-unknown-linux-gnu'; rustupSha256='9029bf864848ab313a2c6806492ae471d1578c8196acdb220e71d6a785241a5a' ;; \
-		i386) rustArch='i686-unknown-linux-gnu'; rustupSha256='4a478c977b7b4900456c2d4dd165019e7c923ebdaba3f47316717d1690387d9a' ;; \
+		amd64) rustArch='x86_64-unknown-linux-gnu'; rustupSha256='f5833a64fd549971be80fa42cffc6c5e7f51c4f443cd46e90e4c17919c24481f' ;; \
+		armhf) rustArch='armv7-unknown-linux-gnueabihf'; rustupSha256='67a98a67f7f7bf19c5cde166499acb8299f2f8fa88c155093df53b66da1f512a' ;; \
+		arm64) rustArch='aarch64-unknown-linux-gnu'; rustupSha256='82fe368c4ebf1683d57e137242793a4417042639aace8bd514601db7d79d3645' ;; \
+		i386) rustArch='i686-unknown-linux-gnu'; rustupSha256='7a1c085591f6c1305877919f8495c04a1c97546d001d1357a7a879cedea5afbb' ;; \
 		*) echo >&2 "unsupported architecture: ${dpkgArch}"; exit 1 ;; \
 	esac; \
     \
-    url="https://static.rust-lang.org/rustup/archive/1.5.0/${rustArch}/rustup-init"; \
+    url="https://static.rust-lang.org/rustup/archive/1.6.0/${rustArch}/rustup-init"; \
     wget "$url"; \
     echo "${rustupSha256} *rustup-init" | sha256sum -c -; \
     chmod +x rustup-init; \
-    ./rustup-init -y --no-modify-path --default-toolchain 1.19.0; \
+    ./rustup-init -y --no-modify-path --default-toolchain 1.20.0; \
     rm rustup-init; \
     chmod -R a+w $RUSTUP_HOME $CARGO_HOME; \
     rustup --version; \
```

</details>

File: library/rust [hunk 1]
```diff
@@ -1,14 +1,24 @@
...
+Architectures: amd64, arm32v7, arm64v8, i386
+GitCommit: bf30ee63bff0a381ccacdb86436bb7da5dc9c853
+Directory: 1.20.0/stretch
+
+Tags: 1.20.0-jessie, 1-jessie, 1.20-jessie, jessie
+Architectures: amd64, arm32v7, arm64v8, i386
+GitCommit: bf30ee63bff0a381ccacdb86436bb7da5dc9c853
+Directory: 1.20.0/jessie
+
+Tags: 1.19.0-stretch, 1.19-stretch, 1.19.0, 1.19
```
Comment (yosifkit) [inline library/rust]: Any particular reason to keep the 1.19 versions listed here?  Removing it here will remove it from the "Supported" section on the Hub readme (and will prevent us from spending cycles rebuilding it on the official build server), but the tag will still be available to users who want it.  (See https://github.com/docker-library/official-images#library-definition-files for more detail on this.)

**Comment by yosifkit** [comment]
Build test of #3400; 7c89f3dd (`rust`):

```console

$ bashbrew build rust:1.20.0-stretch
Building bashbrew/cache:c2f0556e73b58b2b249515843ce4702f1c7af3c04c3c538f3ad7c6254c43b515 (rust:1.20.0-stretch)
Tagging rust:1.20.0-stretch
Tagging rust:1-stretch
Tagging rust:1.20-stretch
Tagging rust:stretch
Tagging rust:1.20.0
Tagging rust:1
Tagging rust:1.20
Tagging rust:latest

$ test/run.sh rust:1.20.0-stretch
testing rust:1.20.0-stretch
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'rust-hello-world' [5/5]...passed


$ bashbrew build rust:1.20.0-jessie
Building bashbrew/cache:c4fa6e8bcb0d25df26ec001e470ee51ba12a67089803e1e62cf4a01f52686714 (rust:1.20.0-jessie)
Tagging rust:1.20.0-jessie
Tagging rust:1-jessie
Tagging rust:1.20-jessie
Tagging rust:jessie

$ test/run.sh rust:1.20.0-jessie
testing rust:1.20.0-jessie
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'rust-hello-world' [5/5]...passed


$ bashbrew build rust:1.19.0-stretch
Building bashbrew/cache:04324413b1fa637cd60ac6b2c52493515753a7cd417513f0bec403f9be1c177b (rust:1.19.0-stretch)
Tagging rust:1.19.0-stretch
Tagging rust:1.19-stretch
Tagging rust:1.19.0
Tagging rust:1.19

$ test/run.sh rust:1.19.0-stretch
testing rust:1.19.0-stretch
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'rust-hello-world' [5/5]...passed


$ bashbrew build rust:1.19.0-jessie
Building bashbrew/cache:0711f0cc753a0a9faf4d22bd47e264b897f8bb97e36037159e7cbdedf02ab121 (rust:1.19.0-jessie)
Tagging rust:1.19.0-jessie
Tagging rust:1.19-jessie

$ test/run.sh rust:1.19.0-jessie
testing rust:1.19.0-jessie
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'rust-hello-world' [5/5]...passed

```

File: library/rust (hunk 1 above)
Comment (sfackler) [reply library/rust]: Ah sure I'll remove them.

Comment (tianon) [reply library/rust]: If there's a chance of 1.19.1 being released (hence, 1.19 still being supported), we don't mind if you keep them, but usually this is just maintainers not being aware of this part of the process. :innocent:

Comment (sfackler) [reply library/rust]: That could happen but it'd be rare - we've only cut a single point release (1.15.1) in the last couple of years. Seems like we can always add a 1.19.1 image set in later if necessary.

Comment (tianon) [reply library/rust]: Indeed! :+1: :heart:

**Pull request merged**